gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/service/JavaServiceDistributionPlugin.java

```diff
@@ -89,9 +90,12 @@ public void apply(Project project) {
             }));
         });
 
+        Configuration runtimeClasspath = project.getConfigurations().getByName("runtimeClasspath");
+        Configuration javaAgentConfiguration = project.getConfigurations().create("javaAgent");
+        javaAgentConfiguration.setTransitive(false);
```

Comment:
It's generally best to configure configurations in an action, so

```java
Configuration javaAgentConfiguration = project.getConfigurations().create("javaAgent", javaAgent -> javaAgent.setTransitive(false));
```

Probably not a huge deal here, but in general this can allow Gradle to avoid doing work when it doesn't need to.

Reply:
Is this one of those things that needs to be an anonymous class? We appear to do that for some other Action types.

Reply:
We have an error prone rule that will warn you (or perhaps a the gradle plugin validation task?). I can never remember the particular cases so I just use a lambda until it complains at me.

Reply:
I believe that you only want an Action when doing a `doFirst`/`doLast` so that the task is properly cached.

I don't think there is performance benefit for configuring a Configuration using a lambda since the configuration is always eagerly created. It is definitely true that for lazy task (i.e. `TaskProvider<?>`) that you want to configure them using a lambda instead of calling `get()` and then configuring the return value